### ISSUE TITLE
[feature] st-util with SWO support possible?

### ISSUE THREAD

**Issue by mhanuel26** [stlink-org/stlink#613]
- [x] Stlink/v2-onboard
- [x] Linux
- [x] `st-util`
- [x] STM32F769 Discovery

I was wondering if it's possible to have SWO output from target. It will be great to have sort of printf window for debug messages. 
The closest minimalist solution I have found is explained [here](https://stm32f4-discovery.net/2014/12/library-46-debug-stm32f4-device-swo-feature/), but depends on STM32 ST-Link Utility or Keil uVision for it.

Please let me know your thoughts,

**Comment by xor-gate** [comment]
Its possible, but not implemented. OpenOCD has support for this to write ITM port 0 or 1 to a file.

**Comment by shizacat** [comment]
Hello,
Do you plan to implement swo?

**Comment by romansavrulin** [comment]
+1 on SWO feature

**Comment by langhaarschneider** [comment]
also +1 on SWO feature

**Comment by simplerobot** [comment]
Hi @mhanuel26, @shizacat, @langhaarschneider,
The stlink project now contains a st-trace utility program that will read SWO output from a target.  (Run "st-trace --help" for more details.)  Does this meet your needs?  If not, can you explain how you would like this to work?

**Comment by Ant-ON** [comment]
@simplerobot Viewing the SWO output while debugging would be a nice touch. Currently only the semihosting is supported:
https://github.com/stlink-org/stlink/blob/01418b5666c72ce802ea5af88b62dc9aa06c8c3e/src/st-util/gdb-server.c#L1423-L1481

**Comment by Nightwalker-87** [comment]
Is there any chance to continue with this?

**Comment by ChrisIdema** [comment]
I really need this feature. I see it is added to the list of release v1.7.1. When can I expect the release of v1.7.1?

**Comment by Nightwalker-87** [comment]
Currently no date or time frame is planned for any new release.
This is because development is currently mostly on hold within this project, see also: https://github.com/stlink-org/stlink/discussions/1223

**Comment by Nightwalker-87** [comment]
I'd consider this a useful feature as well. Who can help out on this topic? Does anybody have an idea how to implement this?

**Comment by Nightwalker-87** [comment]
@Ant-ON Would you be interested in taking action here?

**Comment by ChrisIdema** [comment]
@stlink-org why did you delete my comment?

**Comment by linuxguy123** [comment]
What is the status of SWO working with STLink (st-util) such that Cortex-Debug and uScope (RTT viewer, https://github.com/DEMCON/uscope) can use it ?

If it now does work, does one need to reflash the STLink device to make SWO work or just update st-util ?

If it doesn't work, what can I do to help the situation ?

**Comment by linuxguy123** [comment]
> Hi @mhanuel26, @shizacat, @langhaarschneider, The stlink project now contains a st-trace utility program that will read SWO output from a target. (Run "st-trace --help" for more details.) Does this meet your needs? If not, can you explain how you would like this to work?

Speaking as a VSCode user, how would I configure VSCode to work with st-trace ?  Let's assume I want to use uScope (https://github.com/DEMCON/uscope) as my rtt viewer.

**Comment by Nightwalker-87** [comment]
@linuxguy123 There have been no action or any efforts on this topic since the last response of @Ant-ON.
I personally can't contribute to this topic, sorry.

**Comment by Nightwalker-87** [comment]
Closed due to inactivity and missing interest.
